db/src/models/ticket_instances.rs

```diff
@@ -103,7 +104,7 @@ impl TicketInstance {
             )
             .filter(ticket_instances::id.eq(self.id))
             .select(
-                sql::<Bool>("case when order_items.id is null then true else orders.user_id <> wallets.user_id end")
+                sql::<Bool>("case when order_items.id is null then true else (orders.user_id <> wallets.user_id AND (orders.on_behalf_of_user_id IS NULL OR wallets.user_id <> orders.on_behalf_of_user_id)) end")
```

Comment:
Just some case cleanup, otherwise good find.
```suggestion
                sql::<Bool>("CASE WHEN order_items.id IS NULL THEN TRUE ELSE (orders.user_id <> wallets.user_id AND (orders.on_behalf_of_user_id IS NULL OR wallets.user_id <> orders.on_behalf_of_user_id)) END")
```